server/src/modules/boards/boards.service.ts

```diff
@@ -41,7 +47,45 @@ export class BoardsService {
     boardId: string,
   ): Promise<BoardResponseObject> {
     const deletedBoard = await this.deleteBoardById(boardId);
-    return this.res.toResponseBoard(deletedBoard);
+    return this.res.toResponseBoard(deletedBoard, owner);
+  }
+
+  async getUserRelatedBoards(
+    userId: string,
+    boardsOwnershipFilterOption: BoardsOwnerShipFilterOption,
+    page: number,
+    perPage: number,
+  ): Promise<BoardsPaginatedResponseObject> {
+    const userRelatedBoards: SuperBoardDocument[] = [];
```

Comment:
I think better option would be to fetch from db the exact amount of boards needed. What if user has 100 related boards? Remember - each board keeps slides and slides keep objects as nested docs - it would consume loads of resource. 

You can try with sth like this:

```
@Injectable()
export class MyService {
  constructor(
    @InjectModel(MyDocument.name) private myDocumentModel: Model<MyDocument>,
  ) {}

  async findDocuments(page: number, limit: number): Promise<MyDocument[]> {
    const skip = (page - 1) * limit;
    return await this.myDocumentModel
      .find()
      .skip(skip)
      .limit(limit)
      .exec();
  }
}
```

Keep in mind that you dont need to fetch whole documents, you can specify what fields are needed :) :

 ```
 async findDocuments(page: number, limit: number): Promise<MyDocument[]> {
    const skip = (page - 1) * limit;
    return await this.myDocumentModel
      .find()
      .select('permissions owner')
      .skip(skip)
      .limit(limit)
      .exec();
  }
```